qiita_pet/handlers/study_handlers/description_handlers.py

```diff
@@ -423,6 +423,63 @@ def add_prep_template(self, study, user, callback):
 
         callback((msg, msg_level, 'prep_template_tab', pt_id, None))
 
+    def update_prep_template(self, study, user, callback):
+        """Update a prep template from the POST method
+
+        Parameters
+        ----------
+        study : Study
+            The current study object
+        user : User
+            The current user object
+        callback : function
+            The callback function to call with the results once the processing
+            is done
+
+        Raises
+        ------
+        HTTPError
+            If the prep template file does not exists
+        """
+        # If we are on this function, the arguments "prep_template_id",
+        # "update_prep_template_file" must defined. If not, let tornado
+        # raise its error
+        pt_id = int(self.get_argument('prep_template_id'))
+        prep_template = self.get_argument('update_prep_template_file')
+
+        # Define here the message and message level in case of success
+        msg = "The prep template '%s' has been updated" % prep_template
+        msg_level = "success"
+        # Get the uploads folder
+        _, base_fp = get_mountpoint("uploads")[0]
+        # Get the path of the prep template in the uploads folder
+        fp = join(base_fp, str(study.id), prep_template)
+
+        if not exists(fp):
+            # The file does not exist, fail nicely
+            raise HTTPError(400, "This file doesn't exist: %s" % fp)
```

Comment:
@squirrelo although is a file does not exist error, I don't think this actually fits the 404 error. We don't want the user to be redirected to the 404 page, we want to get him in the study description page and let him know that the file he selected for processing as prep template does not exist. My guess is that this error will be hit almost never, but is better to be safe and put a check here.

Does this makes sense?